src/cmd-cloud-prune

```diff
@@ -414,5 +437,71 @@ def prune_build(s3_client, bucket, prefix, build_id, dry_run):
                 raise Exception(f"Error pruning {build_id}: {e.response['Error']['Message']}")
 
 
+def get_container_tags(meta_json, stream):
+    container_tags = []
+    base_oscontainer = meta_json.get("base-oscontainer")
+    if base_oscontainer:
+        tags = base_oscontainer.get("tags", [])
+        # Only include tags that do not match the stream i.e. moving tags
+        filtered_tags = [tag for tag in tags if tag != stream]
+        if filtered_tags:
+            container_tags = filtered_tags
+    return container_tags
+
+
+def prune_containers(containers_config):
+    barrier_releases = set()
+    # Get the update graph for stable streams
+    if containers_config["stream"] in ['stable', 'testing', 'next']:
+        update_graph = get_update_graph(containers_config["stream"])['releases']
+        # Keep only the barrier releases
+        barrier_releases = set([release["version"] for release in update_graph if "barrier" in release])
```

Comment:
These values won't change during the course of a run, correct?


I feel like this should happen as set up for GC and not deep inside nested for loops. A good way to answer this question is: How many times does `https://builds.coreos.fedoraproject.org/updates/{stream}.json` get hit? Is it way more than necessary? If so, refactor.

I also think we shouldn't consider barrier releases inside this code at all. we should bail out at the very beginning of the `case "containers":` if it's a barrier release or am I missing something?

Reply:
resolved in new upload